### DISCUSSION TITLE
Escape to clear code at console prompt

### DISCUSSION THREAD

**Discussion by srhrnkn** [posit-dev/positron#5186] [Feedback] [answered]
In RStudio, the escape key clears any text at the console prompt. In Positron the escape key doesn't do this (or anything else) if the cursor is at the console prompt. Is there any way to change Positron to the RStudio behavior?

(Interestingly I just opened a naked R session and there hitting escape seems to give you a new prompt line without executing the code, which is confusing.)

**Comment by benzipperer** [comment]
I agree, ESC would be great. However, for now it appears CTRL-C gives you a new prompt line without executing the code.

**Reply by srhrnkn** [reply]
Good tip, thanks!

**Comment by lionel-** [comment, marked as answer]
We intentionally removed this behaviour because escape is also used to close popups. It's easy to hit escape one time too many and get your input accidentally cleared out.

In addition to Ctrl-C you can also use Ctrl-U. Both of these also work in terminals, which is a nice consistency. Or Ctrl-A / Cmd-A to select all the input and get the selection replaced by the next character you type.

**Reply by srhrnkn** [reply]
Makes sense, thanks - one more brain remapping task

**Comment by moa-incom** [comment]
I just wanted to add a little bit to the discussion here in case anyone else is looking for how to interrupt or cancel the running console execution. I understand the reasoning behind the removal of this RStudio feature as it has caused countless unintentional interruptions of running code. However, for me, it was a nightmare to find information about how to interrupt the running code in the console in Positron - until this post; `CTRL + C` also works for this!
It took me way too long to find this than it should have. Perhaps better information about the console actions could be communicated/displayed. Keyboard Shortcut naming also didn't really give me any leads.

**Comment by ruivicente2** [comment]
And from a running script?

- cat("\f") do not work neither cat("\n").

thanks -- good work

🤐
